jsk_unitree_robot/jsk_unitree_startup/scripts/walk_notifier.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding:utf-8 -*-
+
+import os
+import subprocess
+import time
+import re
+import json
+import tempfile
+
+import PIL.Image
+import sensor_msgs.msg
+from std_srvs.srv import Trigger
+import rospy
+import cv_bridge
+
+
+def send_mail(place, robot_name, sender_address, receiver_address, attachment=None):
+    """
+    Send mail with mailutils
+    """
+    mail_title = u"{}、お散歩中です。".format(robot_name)
+    message = u"お散歩してます。"
+    if place is not None:
+        message += u"\n今は{}を歩いているよ。".format(place)
+    cmd = u"mail -s '{}' -r {} {}".format(
+        mail_title, sender_address, receiver_address)
+    if attachment is not None:
+        cmd += ' -A {}'.format(attachment)
+    cmd = ['mail', '-s', mail_title.encode('utf-8'),
+           '-r', sender_address,
+           receiver_address]
+    if attachment is not None:
+        cmd += ['-A', attachment]
+    rospy.loginfo('Executing: {}'.format(' '.join(cmd)))
+    process = subprocess.Popen(cmd,
+                               stdin=subprocess.PIPE,
+                               universal_newlines=True)
+    process.communicate(message.encode('utf-8'))
+    process.wait()
+
+
+class WalkNotifier(object):
+
+    def __init__(self):
+        self.bridge = cv_bridge.CvBridge()
+        self.robot_name = rospy.get_param('~robot_name', 'unitree').strip()
+        self.sub = rospy.Subscriber('~input_image',
+                                    sensor_msgs.msg.Image,
+                                    callback=self.callback,
+                                    queue_size=1)
+        self.sender_address = rospy.get_param('~sender_address')
+        self.receiver_address = rospy.get_param('~receiver_address')
+        self.notify_interval = int(rospy.get_param('~notify_interval', 180))
+        rospy.wait_for_service('~get_location')
+        self.wait_image()
+
+    def callback(self, img_msg):
+        self.img = self.bridge.imgmsg_to_cv2(img_msg, desired_encoding='bgr8')
+
+    def wait_image(self):
+        self.img = None
+        rate = rospy.Rate(10)
+        while not rospy.is_shutdown() and self.img is None:
+            rate.sleep()
+            rospy.loginfo('[WalkMail] waiting image')
+
+    def get_place(self):
```

Comment:
please consider using https://github.com/jsk-ros-pkg/jsk_robot/pull/1558, because as you named your node as `notifier` , so sending E-mail is not important, but to send notification is important.
Current implementation start storing smach description from START. And send email when SMACH goes to END state.
So if we want to send email periodically, we need to write smach like

```mermaid
flowchart TB
  INIT --> START
  START --> GET-LOCATION
  GET-LOCATION  --> END
  END --> WALK
  WALK --> START
  END --> TERMINATE
``` 
https://github.com/k-okada/jsk_robot/blob/pepper_cross/jsk_robot_common/jsk_robot_startup/lifelog/speaking-program-is-started-or-terminated.l

Or, we could update https://github.com/jsk-ros-pkg/jsk_robot/pull/1558 to select send notification every state or not.

Reply:
Thanks! 
I created `walk_notifier.l` by using smach.
https://github.com/iory/jsk_robot/blob/unitree-camera-walk-app/jsk_unitree_robot/jsk_unitree_startup/scripts/walk-notifier.l